lib/chibi/test.scm

```diff
@@ -521,39 +527,34 @@
         info)))
 
 (define (test-default-applier expect expr info)
-  (let* ((group (current-test-group))
-         (indent (and group (test-group-indent-width group))))
-    (cond
-     ((or (not group) (test-group-ref group 'verbose))
-      (if (and indent (positive? indent))
-          (display (make-string indent #\space)))
-      (test-print-name info indent)))
-    (let ((expect-val
-           (guard
-               (exn
-                (else
-                 (warning "bad expect value")
-                 (print-exception exn (current-error-port))
-                 #f))
-             (expect))))
-      (guard
-          (exn
-           (else
-            ((current-test-reporter)
-             (if (assq-ref info 'expect-error) 'PASS 'ERROR)
-             (append `((exception . ,exn)) info))))
-        (let ((res (expr)))
-          (let ((status
-                 (if (and (not (assq-ref info 'expect-error))
-                          (if (assq-ref info 'assertion)
-                              res
-                              ((current-test-comparator) expect-val res)))
-                     'PASS
-                     'FAIL))
-                (info `((result . ,res) (expected . ,expect-val) ,@info)))
-            ((current-test-reporter) status info)))))))
+  ((current-test-reporter) #f info)
```

Comment:
It seems weird to pass #f here to indicate a completely different semantics (indicating the start of a test) for the test reporter.  Better to split this into a separate parameter if needed.

Reply:
I used the special symbol 'BEGIN to indicate that a test is about to begin. For simplicity this is passed also for skipped tests. I did not use a separate parameter to keep the interface stable.